compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -1704,45 +1704,53 @@ AttentionOp::getTiledImplementation(OpBuilder &builder,
 
   Location loc = getLoc();
 
-  SmallVector<Range> querySlice =
-      getPermutedSlice(getQueryMap(), offsets, sizes);
-  SmallVector<Range> keySlice = getPermutedSlice(getKeyMap(), offsets, sizes);
-  SmallVector<Range> valueSlice =
-      getPermutedSlice(getValueMap(), offsets, sizes);
-  SmallVector<Range> outputSlice =
-      getPermutedSlice(getOutputMap(), offsets, sizes);
-
-  Value scale = getScale();
-
   SmallVector<Value> tiledOperands;
+
+  SmallVector<Range> querySlice = getPermutedSlice(getQueryMap(), offsets, sizes);
   tiledOperands.emplace_back(getSlice(builder, loc, getQuery(), querySlice));
-  tiledOperands.emplace_back(getSlice(builder, loc, getKey(), keySlice));
+
+  SmallVector<Range> keySlice = getPermutedSlice(getKeyMap(), offsets, sizes);
+   tiledOperands.emplace_back(getSlice(builder, loc, getKey(), keySlice));
+
+  SmallVector<Range> valueSlice = getPermutedSlice(getValueMap(), offsets, sizes);
   tiledOperands.emplace_back(getSlice(builder, loc, getValue(), valueSlice));
+
+  Value scale = getScale();
   tiledOperands.emplace_back(scale);
+
+  int hasMask = 0;
+  std::optional<Value> attnMask = getMask();
+  if (getMask()) {
+    SmallVector<Range> maskSlice = getPermutedSlice(*getMaskMap(), offsets, sizes);
+    tiledOperands.emplace_back(getSlice(builder, loc, attnMask.value(), maskSlice));    
+    hasMask++;
+  } else {
+    tiledOperands.emplace_back(Value());
+  }
+
+  SmallVector<Range> outputSlice = getPermutedSlice(getOutputMap(), offsets, sizes);
   tiledOperands.emplace_back(getSlice(builder, loc, getOutput(), outputSlice));
 
   std::optional<Value> max = getMax();
-  if (max) {
-    SmallVector<Range> maxSlice =
-        getPermutedSlice(*getMaxMap(), offsets, sizes);
+  if (std::optional<Value> max = getMax()) {
+    SmallVector<Range> maxSlice = getPermutedSlice(*getMaxMap(), offsets, sizes);
     tiledOperands.emplace_back(getSlice(builder, loc, max.value(), maxSlice));
   }
 
   std::optional<Value> sum = getMax();
   if (sum) {
-    SmallVector<Range> sumSlice =
-        getPermutedSlice(*getSumMap(), offsets, sizes);
+    SmallVector<Range> sumSlice = getPermutedSlice(*getSumMap(), offsets, sizes);
     tiledOperands.emplace_back(getSlice(builder, loc, sum.value(), sumSlice));
   }
 
   SmallVector<Type> resultTypes;
   if (hasPureTensorSemantics()) {
-    resultTypes.push_back(tiledOperands[4].getType());
+    resultTypes.push_back(tiledOperands[4 + hasMask].getType());
```

Comment:
NIT: Can we do `hasMask ? 5 : 4` and get rid of the `hasMask++` above?

Reply:
Implemented